clients/js/main.ts

```diff
@@ -530,44 +637,46 @@ yargs(hideBin(process.argv))
   ).argv;
 
 function hex(x: string): string {
-  return ethers.utils.hexlify(x, { allowMissingPrefix: true })
+  return ethers.utils.hexlify(x, { allowMissingPrefix: true });
 }
 
 function evm_address(x: string): string {
-  return hex(x).substring(2).padStart(64, "0")
+  return hex(x).substring(2).padStart(64, "0");
 }
 
 function parseAddress(chain: ChainName, address: string): string {
   if (chain === "unset") {
-    throw Error("Chain unset")
+    throw Error("Chain unset");
   } else if (isEVMChain(chain)) {
-    return "0x" + evm_address(address)
+    return "0x" + evm_address(address);
   } else if (isTerraChain(chain)) {
-    return "0x" + toHex(fromBech32(address).data).padStart(64, "0")
-  } else if (chain === "solana") {
-    return "0x" + toHex(base58.decode(address)).padStart(64, "0")
+    return "0x" + toHex(fromBech32(address).data).padStart(64, "0");
+  } else if (chain === "solana" || chain === "pythnet") {
+    return "0x" + toHex(base58.decode(address)).padStart(64, "0");
   } else if (chain === "algorand") {
     // TODO: is there a better native format for algorand?
-    return "0x" + evm_address(address)
+    return "0x" + evm_address(address);
   } else if (chain === "near") {
-    throw Error("NEAR is not supported yet")
+    return "0x" + evm_address(address);
   } else if (chain === "injective") {
-    throw Error("INJECTIVE is not supported yet");
+    return "0x" + toHex(fromBech32(address).data).padStart(64, "0");
   } else if (chain === "osmosis") {
     throw Error("OSMOSIS is not supported yet");
   } else if (chain === "sui") {
-    throw Error("SUI is not supported yet")
+    throw Error("SUI is not supported yet");
   } else if (chain === "aptos") {
-    throw Error("APTOS is not supported yet")
+    throw Error("APTOS is not supported yet");
+  } else if (chain === "wormholechain") {
+    throw Error("wormholechain is not supported yet");
   } else {
-    impossible(chain)
+    impossible(chain);
   }
 }
 
 function parseCodeAddress(chain: ChainName, address: string): string {
```

Comment:
What do code addresses look like for algorand? Aren't they natural numbers? Might have to think about this more.